geomstats/learning/wrapped_gaussian_process.py

```diff
@@ -0,0 +1,290 @@
+    regression on riemannian manifolds. In 2018 IEEE/CVF
+    Conference on Computer Vision and Pattern Recognition
+    """
+
+    def __init__(
+        self,
+        space,
+        metric,
+        prior,
+        kernel=None,
+        *,
+        alpha=1e-10,
+        optimizer="fmin_l_bfgs_b",
+        n_restarts_optimizer=0,
+        copy_X_train=True,
+        random_state=None,
+    ):
+        if metric is None:
+            metric = space.metric
+        self.metric = metric
+        self.space = space
+        self.prior = prior
+        self.copy_X_train = copy_X_train
+
+        self.y_train_ = None
+        self.tangent_y_train_ = None
+
+        self._euclidean_gpr = GaussianProcessRegressor(
+            kernel=kernel,
+            alpha=alpha,
+            optimizer=optimizer,
+            n_restarts_optimizer=n_restarts_optimizer,
+            normalize_y=False,
+            copy_X_train=copy_X_train,
+            random_state=random_state,
+        )
+
+        # I know that inheritance seems more appropriate here, but the issue is
+        # that the .sample_y method of wgrp calls the .sample_y of gpr,
+        # which calls the .predict of wgpr instead of the one of gpr.
+        # Moreover the attribute .y_train_ would be the tangent_y_train and not the true
+        # y_train.
+
+        self.__dict__.update(self._euclidean_gpr.__dict__)
+        self.log_marginal_likelihood = self._euclidean_gpr.log_marginal_likelihood
+
+    def _get_tangent_targets(self, X, y):
+        """Compute the tangent targets, using the provided prior.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features) or list of object
+            Feature vectors or other representations of training data.
+        y : array-like of shape (n_samples,) or (n_samples, n_targets)
+            Target values. The target must belongs to the manifold space
+
+        Returns
+        -------
+        tangent_y : array-like of shape (n_samples,) or (n_samples, n_targets)
+                Target projected on the associated (by the prior) tangent space.
+        """
+        base_points = self.prior(X)
+        return self.metric.log(y, base_point=base_points)
+
+    def fit(self, X, y):
+        """Fit Wrapped Gaussian process regression model.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features) or list of object
+            Feature vectors or other representations of training data.
+        y : array-like of shape (n_samples,) or (n_samples, n_targets)
+            Target values. The target must belongs to the manifold space
+
+        Returns
+        -------
+        self : object
+            WrappedGaussianProcessRegressor class instance.
+        """
+        if not gs.all(self.space.belongs(y)):
+            raise AttributeError("The target values must belongs to the given space")
+
+        # compute the tangent dataset using the prior
```

Comment:
Could you move these comments into the description of the docstring of `fit`? You can write something like:

The Wrapped Gaussian process is fit through the following steps:
 1. Compute the tangent dataset using the prior
 2. Fit a Gaussian process regression on the tangent dataset
 3. ...

Putting this in the docstring allows it to show up in the documentation website, and it is still visible from the code.